docs/proposals/20210402-unitedobject.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Proposal Template
+authors:
+- "@hoopoe61"
+reviewers:
+- "@jiuzhu"
+creation-date: 2021-04-02
+last-updated: 2021-04-20
+status: implementable
+---
+
+# UnitedObject: describe similar k8s objects unitedly
+
+Motivation:
+  In now k8s implementation，similar objects are needed to be created sperately. This way produces duplicated and distributed content in k8s, which increases maintenance cost.
+
+Goal:
+  A common way is required to abstract and encapsulate similar objects in k8s, so objects with slight difference can be described unitedly in one object.
+
+Proposal:
+  This proposal try to abstract and encapsulate similar objects in k8s.
+  1. Adding a new CRD and controller for UnitedObject.
+  2. UnitedObject can be used to abstract all kinds of k8s objects.
+  3. Use jsonPatch to describe the differences among objects in one UnitedObject.
+
+Use case
+  1. k8s objects with similar attributes, for example:
```

Comment:
please enrich the user cases， and clarify following questions
1. why duplicated content will increase maintenance cost ？ 
2. why united object can decrease  maintenance cost 
3. are there any other solutions to tackle the problem, such as kustomize/helm, what is the relative pros and cons?

Reply:
1. If the user creates multiple objects separately, the user have to list and inspect all objects to checkout related objects. This is not friendly for user.
2. With the unitedobject, the user can only focus on one object and get all needed info in one object. Maintenance among sub objects will be handled by unitedobject controller. 
3. kustomize/helm is used to simplify the deploy of k8s object, and the unitedobject, similar with uniteddeployment, is used to gather objects in one place. So unitedobject is better for update and maintain objects. 
4. Other use cases such as jobs with different args in command, pod with different volumemounts, pv with different mountOptions or path. 

I think unitedobject provides a common way to describe similar objects in one place. So users can define one object for any resource in k8s and customize fields(expect gvk) according to their requirements.